### ISSUE TITLE
Cannot assign to read only property 'constructor' after updating to v3.2.x

### ISSUE THREAD

**Issue by albertogasparin** [testdouble/testdouble.js#271]
I'm using `td.replace` to replace methods of a Koa router:

```js
let router = new koaRouter();
td.replace(router, 'get');
```

The code above works fine with testdouble 3.1.x however after updating to 3.2.1 it breaks with the following error:

```
Cannot assign to read only property 'constructor' of object '[object Generator]'
      at exports.default (node_modules/testdouble/lib/imitate/overwrite-children/chain-prototype.js:17:27)
      at node_modules/testdouble/lib/imitate/overwrite-children/index.js:38:43
      at newValue (node_modules/testdouble/lib/imitate/overwrite-children/copy-props.js:39:20)
      at node_modules/testdouble/lib/imitate/overwrite-children/copy-props.js:18:28
      at node_modules/lodash/transform.js:60:12
      at node_modules/lodash/_createBaseFor.js:17:11
      at baseForOwn (node_modules/lodash/_baseForOwn.js:13:20)
      at Object.transform (node_modules/lodash/transform.js:59:39)
      at exports.default (node_modules/testdouble/lib/imitate/overwrite-children/copy-props.js:14:52)
      at exports.default (node_modules/testdouble/lib/imitate/overwrite-children/index.js:37:29)
      at imitate (node_modules/testdouble/lib/imitate/index.js:33:35)
      at getFake (node_modules/testdouble/lib/replace/property.js:57:34)
      at exports.default (node_modules/testdouble/lib/replace/property.js:15:37)
      at exports.default [as replace] (node_modules/testdouble/lib/replace/index.js:11:31)
      at Context.beforeEach (api/video/__tests__/unit.index.js:21:12)
      at Hook.Runnable.run (node_modules/co-mocha/lib/co-mocha.js:43:16)
```

Node: v6.10.3
Testdouble: v3.2.1

**Comment by searls** [comment]
Could you reproduce this on runkit or a sample app for me? I tried and failed. Here's the runkit url i used:
 
https://runkit.com/searls/testdouble-js---271

**Comment by albertogasparin** [comment]
My bad, I should have written a better example. I'm using an older version of koa-router (version 7 does not use generators anymore). This is the smallest scenario I can think of:

```js
var td = require("testdouble")

var test = {
  *method () { return 'ok' }
};

var fakeMethod = td.replace(test, 'method');
```

(also failing on [Runkit](https://runkit.com/albertogasparin/595214a9c653e7001260e12a))

**Comment by searls** [comment]
Thanks, that'll do it! As you might surmise I didn't consider generators.

**Comment by searls** [comment]
This was fixed in #274 & landed in 3.2.2
